src/components/Images.js

```diff
@@ -1,154 +1,562 @@
+      );
+      const { thumbnails } = this.state;
+      thumbnails[i] = canvas.toDataURL("image/jpeg");
+      this.setState({ thumbnails });
+    };
+  };
+  generateThumbnail = async (url, i) => {
+    let canvas = document.createElement("canvas");
+    canvas.width = 640;
+    canvas.height = 480;
+    canvas.crossOrigin = "anonymous";
+    let context = canvas.getContext("2d");
+    let video = document.createElement("video");
+    video.width = 0;
+    video.id = `video-${i}`;
+    video.crossOrigin = "anonymous";
+    let source = document.createElement("source");
+    source.src = url;
+    video.appendChild(source);
+    document.getElementsByTagName("html")[0].append(video);
+    video.autoplay = true;
+    video.muted = true;
+    video.ontimeupdate = () => {
+      video.pause();
+      // alert("asdas");
+      context.drawImage(
+        document.getElementById(`video-${i}`),
+        // video,
+        0,
+        0,
+        canvas.width,
+        canvas.height
+      );
+      const { thumbnails } = this.state;
+      thumbnails[i] = canvas.toDataURL("image/jpeg");
+      this.setState({ thumbnails });
+    };
+  };
+  componentDidMount() {
+    const { images } = this.props;
+    if (!images.length) return;
+
+    const imageUrls = [];
+    const thumbnails = [];
+
+    images.forEach((img, i) => {
+      if (this.pureTypeOf(img) === "object") {
+        if ("iFrame" in img) {
+          imageUrls.push(() => (
+            <iframe
+              title="cats"
+              width="560"
+              height="315"
+              src={img.url}
+              style={{
+                maxWidth: "97%",
+                position: "absolute",
+                left: 0,
+                right: 0,
+                margin: "auto",
+                top: "50%",
+                transform: "translateY(-50%)"
+              }}
+              {...img.props}
+            />
+          ));
+          thumbnails.push(img.thumbnail || null);
+        } else if ("isVideo" in img && img.isVideo) {
+          imageUrls.push(() => (
+            <div
+              style={{
+                width: "100%",
+                height: "100%",
+                display: "flex",
+                flex: 1,
+                justifyContent: "center",
+                alignItems: "center"
+              }}
+            >
+              <video
+                width="400"
+                controls
+                source={img.url}
+                {...img.props}
+                crossOrigin="anonymous"
+              >
+                <source src={img.url} />
+              </video>
+            </div>
+          ));
+          thumbnails.push(img.thumbnail || this.generateThumbnail(img.url, i));
+        } else {
+          imageUrls.push(img.url);
+          thumbnails.push(img.thumbnail || img.url);
+        }
+      } else {
+        imageUrls.push(img);
+        thumbnails.push(img);
+      }
+
+      this.setState({ imageUrls, thumbnails });
+    });
+  }
 
```

Comment:
@hamsahmedansari we shouldn't work directly in componentDidMount, create a function and call it from `componentDidMount`